libqtile/backend/x11/core.py

```diff
@@ -771,9 +775,17 @@ def handle_MapRequest(self, event) -> None:  # noqa: N802
             self.update_client_lists()
             win.change_layer()
 
+    def handle_CreateNotify(self, event) -> None:  # noqa: N802
+        xwin = window.XWindow(self.conn, event.window)
+        win = window._Window(xwin, self.qtile)
```

Comment:
We use the `_Window` class as `Window` sets various attributes etc on `__init__` which results in errors for some windows.

Reply:
I wonder if we even need a _Window? Can we keep a cache of wids here we're not supposed to touch? Seems like we should also check for the OverrideRedirect bitmask right?

Reply:
Or even better, no cache at all, and if we see events for wids we don't know about, we could GetWindowAttributes for those and check the OverrideRedirect flag?

Reply:
OK. Help me out a bit here. How did you imagine this working?

If we listen for messages in `handle_ClientMessage`, these arrive before windows are mapped and I suspect we don't really want to get attributes every time we get a message.

Also, from a quick look, some override redirect windows (like VLC's menu) are not transients so I guess we just bring these to the front as they're closed if they lose focus. However, we'll need to track transients so that they can be stacked with their parent if other windows are stacked above them.

Reply:
> I suspect we don't really want to get attributes every time we get a message

I was thinking we would, if only to avoid the two-sources-of-truth problem. Once we keep a cache, we have to remember to update it everywhere that it could be invalidated, and I suspect that will be hard.

But it's not really every time we get a message is it? We "just" want to skip over replies from QueryTree in change_layer() that are managed on their own; given that we're already doing a request for every window in the QueryTree reply to potentially adjust its stacking, I don't think it's that expensive?

We don't get any other events for these windows, so we won't accidentally touch them anywhere but here, IIUC.

> Also, from a quick look, some override redirect windows (like VLC's menu) are not transients so I guess we just bring these to the front as they're closed if they lose focus.

I was hoping (heh, famous last words) that just not modifying their stacking would be enough here. But maybe I'm wrong, I haven't played with any of this in actuality :)

Reply:
OK. You're going to need to spell this out for me. Are we listening for client messages or are we listening for something else? If it's not a managed window, check if it's transient and then move to top/above its parent as necessary?

Reply:
I was thinking of this: https://github.com/qtile/qtile/commit/cc122253f5d7e76047869f213113b5963e79ffa4

But I don't know if that will really work. I guess the question is: if an OverrideRedirect window for an application that has been modified with StackMode.Below some other window pops up, will the application map that window at the same stacking level, or above the other window? And really, shouldn't that OverrideRedirect window probably be on top regardless of what window is actually on top, since it's presumably a dialog or whatever?

But maybe this is all my perspective. I don't use floating much, so any window that's floating for me is probably some kind of modal thing that I want to know about.